CRM/Core/Payment/Dummy.php

```diff
@@ -137,6 +137,32 @@ protected function supportsLiveMode() {
     return TRUE;
   }
 
+  /**
+   * Does this payment processor support refund?
+   *
+   * @return bool
+   */
+  public function supportsRefund() {
+    return TRUE;
+  }
+
+  /**
+   * Submit a refund payment
+   *
+   * @param array $params
+   *   Assoc array of input parameters for this transaction.
+   */
+  public function doRefund(&$params) {
+    if (empty($params['trxn_id'])) {
+      throw new PaymentProcessorException('Refund failed');
+    }
+    else {
+      $params['payment_status_id'] = CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Refunded');
```

Comment:
Can we just keep this function empty for now - I think our discussions so far have been that we are not agreed on whether we should be setting payment_status_id. We'll need to start adding tests etc but I think until we build out the next level an empty function with the comment block stating that if no exception is thrown it is understood to be a success